metricbeat/module/linux/memory/data.go

```diff
@@ -18,51 +18,56 @@
 package memory
 
 import (
+	"os"
+	"strconv"
+	"strings"
+
 	"github.com/pkg/errors"
 
 	"github.com/elastic/beats/v7/libbeat/common"
 	"github.com/elastic/beats/v7/libbeat/metric/system/resolve"
 	"github.com/elastic/beats/v7/metricbeat/internal/metrics/memory"
-	sysinfo "github.com/elastic/go-sysinfo"
-	sysinfotypes "github.com/elastic/go-sysinfo/types"
 )
 
 // FetchLinuxMemStats gets page_stat and huge pages data for linux
 func FetchLinuxMemStats(baseMap common.MapStr, hostfs resolve.Resolver) error {
-
-	vmstat, err := GetVMStat()
+	vmstat, err := GetVMStat(hostfs)
 	if err != nil {
 		return errors.Wrap(err, "error fetching VMStats")
 	}
 
-	pageStats := common.MapStr{
-		"pgscan_kswapd": common.MapStr{
-			"pages": vmstat.PgscanKswapd,
-		},
-		"pgscan_direct": common.MapStr{
-			"pages": vmstat.PgscanDirect,
-		},
-		"pgfree": common.MapStr{
-			"pages": vmstat.Pgfree,
-		},
-		"pgsteal_kswapd": common.MapStr{
-			"pages": vmstat.PgstealKswapd,
-		},
-		"pgsteal_direct": common.MapStr{
-			"pages": vmstat.PgstealDirect,
-		},
+	pageStats := common.MapStr{}
+	if pages, ok := vmstat["pgscan_kswapd"]; ok {
+		pageStats.Put("pgscan_kswapd.pages", pages)
```

Comment:
All of these if blocks would be better as a helper function, to avoid the chance of a typo between the map key and the argument to the `Put` call.

Reply:
Yah, agreed.